enterprise_access/apps/bffs/tests/test_context.py

```diff
@@ -1,40 +1,82 @@
 """
 Text for the BFF context
 """
+import ddt
 from django.test import RequestFactory, TestCase
+from faker import Faker
 from rest_framework.exceptions import ValidationError
 
-from enterprise_access.apps.api_client.tests.test_constants import DATE_FORMAT_ISO_8601
 from enterprise_access.apps.bffs.context import HandlerContext
 from enterprise_access.apps.core.tests.factories import UserFactory
-from enterprise_access.utils import _curr_date
 
 
+@ddt.ddt
 class TestHandlerContext(TestCase):
     def setUp(self):
         super().setUp()
         self.factory = RequestFactory()
         self.mock_user = UserFactory()
+        self.faker = Faker()
+
+        self.mock_enterprise_customer_uuid = self.faker.uuid4()
+        self.request = self.factory.get('sample/api/call')
+        self.request.user = self.mock_user
+        self.request.query_params = {
+            'enterprise_customer_uuid': self.mock_enterprise_customer_uuid
+        }
+        self.request.data = {
+            'enterprise_customer_uuid': self.mock_enterprise_customer_uuid
+        }
```

Comment:
nit: I'm not sure if it makes sense to initialize the mock request with **both** `query_params` AND `data` simultaneously, as typically it'd be either one or the other.

Reply:
You are correct but there is a test that checks for whether the query params or data is included. Selectively removing them from the request from `setUp` makes for a cleaner test.
``` py
  @ddt.data(
        {
            'query_params': True,
            'data': True,
        },
        {
            'query_params': False,
            'data': True,
        },
        {
            'query_params': True,
            'data': False,
        },
        {
            'query_params': False,
            'data': False,
        }
    )
    @ddt.unpack
    def test_handler_context_enterprise_customer_uuid(self, query_params, data):
        if not query_params:
            del self.request.query_params['enterprise_customer_uuid']
        if not data:
            del self.request.data['enterprise_customer_uuid']
        request = self.request
        if not (query_params or data):
            with self.assertRaises(ValueError):
                HandlerContext(request)
        else:
            context = HandlerContext(request)
            self.assertEqual(context.enterprise_customer_uuid, self.mock_enterprise_customer_uuid)
            self.assertEqual(context.lms_user_id, self.mock_user.lms_user_id)
 ```

Reply:
So I made a slight refactor to determine whether a `request_object` exist to parse out the `enterprise_customer_uuid`. Reason being was the original tuple was failing on retrieving the `data` object if neither existed.

It determines which `request_object` exist, either a `query_parameter` or `data` object, and throws a `ValueError` if neither exist. If a `request_object` exist, then we parse out the `enterprise_customer_uuid` and determine it does exist. If not, throw a `KeyError` that a enterprise_customer_uuid is required.

To the point of a data and query_param should not exist in tandem, it is still covered in the tests for full coverage to validate that we handle it correctly.